compact.go

```diff
@@ -757,7 +757,16 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 		}
 
 		for i, chk := range chks {
-			if chk.MinTime < meta.MinTime || chk.MaxTime > meta.MaxTime {
+			// Re-encode head chunks that are still open (being appended to) or
+			// outside the compacted MaxTime range.
+			// The chunk.Bytes() method is not safe for open chunks hence the re-encoding.
+			// This happens when snapshotting the head block.
+			//
+			// Bock time range is half-open: [meta.MinTime, meta.MaxTime) and
+			// chunks are closed hence the chk.MaxTime >= meta.MaxTime not check.
+			if _, isHeadChunk := chk.Chunk.(*safeChunk); isHeadChunk && chk.MaxTime >= meta.MaxTime {
+				dranges = append(dranges, Interval{Mint: meta.MaxTime, Maxt: math.MaxInt64})
+			} else if chk.MinTime < meta.MinTime || chk.MaxTime >= meta.MaxTime { // Sanity check for disk blocks.
```

Comment:
` >= ` Wonder if this will not break compaction for old users who might have such chunk for some reaons :thinking:

Reply:
shouldn't but maybe warning would be nice instead here? (sorry it was my idea to put it)

Reply:
Especially that there might not much harm in leaving just `>`

Reply:
updated to be only a warning.

Reply:
Why are we not returning error here? Wouldn't this mean we would allow chunks outside the range?

Reply:
We should still have an error here

Reply:
> shouldn't but maybe warning would be nice instead here? (sorry it was my idea to put it)

I meant warning for `ch.MaxTime == meta.MaxTime` and error for the rest... sorry for confusion - should be more explicit here ):

Reply:
We should definitely not add regression and not skip the sanity error to what we had previously

Reply:
but why the hard fail? The next compaction would remove these extra samples from the chunks, no?

Reply:
This would lead to a mismatch in what the meta says and what is actually in the block if we allow chunks outside the meta range.